custom/icds_reports/reports/take_home_ration.py

```diff
@@ -97,3 +82,115 @@ def _format_infrastructure_data(data):
                 filters
             ]
         ]
+
+    def get_consolidated_data(self, filters, order_by):
+        headers = ['State', 'District', 'Block', 'Sector', 'Awc Name', 'AWW Name', 'AWW Phone No.',
+                   'Total No. of Beneficiaries eligible for THR',
+                   'Total No. of beneficiaries received THR in given month',
+                   'Total No of Pictures taken by AWW']
+
+        if self.beta:
+            thr_column = 'thr_21_days' if self.month <= datetime(2020, 3, 1).date() else 'thr_25_days'
+            launched_column = 'num_launched_awcs'
+            thr_eligible_column = 'thr_eligible'
+            class_model = ServiceDeliveryReportView
+        else:
+            thr_column = 'thr_given_21_days'
+            launched_column = 'is_launched'
+            thr_eligible_column = 'total_thr_candidates'
+            class_model = TakeHomeRationMonthly
+
+        columns = [
+            'state_name', 'district_name', 'block_name',
+            'supervisor_name', 'awc_name', 'aww_name', 'contact_phone_number',
+            launched_column,
+            thr_eligible_column,
+            thr_column,
+            'thr_distribution_image_count'
+        ]
+
+        query_set = class_model.objects.filter(**filters).order_by(*order_by)
+        data = query_set.values(*columns)
+        return headers, data
+
+    def get_beneficiary_wise_data(self, filters, order_by):
+
+        thr_days = '21' if self.month <= datetime(2020, 3, 1).date() else '25'
```

Comment:
Please add a comment why are we taking 21 before that date and 25 after that date (2020-03-01) and set this as constant in the class constructor function and any specific reason of not directly using `date(2020,3,1)`